android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/multiinsert/DataStoreSingleInsertTest.kt

```diff
@@ -0,0 +1,223 @@
+    }
+
+    @Test
+    @Throws(InterruptedException::class)
+    fun testTempIdAuto() {
+        testTempId(StoreType.AUTO)
+    }
+
+    @Test
+    @Throws(InterruptedException::class)
+    fun testTempIdNetwork() {
+        testTempId(StoreType.NETWORK)
+    }
+
+    @Throws(InterruptedException::class)
+    private fun testTempId(storeType: StoreType) {
+        val person = createPerson(TEST_USERNAME)
+        val personStore = DataStore.collection(MULTI_INSERT_COLLECTION, Person::class.java, storeType, client)
+        clearBackend(personStore)
+        client.syncManager.clear(MULTI_INSERT_COLLECTION)
+        val saveCallback = create(personStore, person)
+        assertNull(saveCallback.error)
+        assertNotNull(saveCallback.result)
+        if (storeType == StoreType.SYNC) {
+            Assert.assertTrue(saveCallback.result!!.id?.startsWith("temp")!!)
+            val pushCallback = push(personStore, DEFAULT_TIMEOUT)
+            assertNull(pushCallback.error)
+            assertNotNull(pushCallback.result)
+        } else {
+            Assert.assertTrue(!saveCallback.result!!.id?.startsWith("temp")!!)
+        }
+        val findCallback = find(personStore, LONG_TIMEOUT)
+        assertNotNull(findCallback.result)
+        assertNotNull(findCallback.result!!.result)
+        if (storeType == StoreType.SYNC) {
+            Assert.assertTrue(saveCallback.result!!.id?.startsWith("temp")!!)
+        } else {
+            Assert.assertTrue(!saveCallback.result!!.id?.startsWith("temp")!!)
+        }
+
+    }
+
+    @Test
+    @Throws(InterruptedException::class)
+    fun testErrorMessageIfSameIdExistsNetwork() {
+        testErrorMessageIfSameIdExists(StoreType.NETWORK)
+    }
+
+    @Test
+    @Throws(InterruptedException::class)
+    fun testErrorMessageIfSameIdExistsAuto() {
+        testErrorMessageIfSameIdExists(StoreType.AUTO)
+    }
+
+    @Test
+    @Ignore("Should work after fixing: https://kinvey.atlassian.net/browse/KDEV-781")
+    @Throws(InterruptedException::class)
+    fun testErrorMessageIfSameIdExistsSync() {
+        testErrorMessageIfSameIdExists(StoreType.SYNC)
+    }
+
+    @Throws(InterruptedException::class)
+    fun testErrorMessageIfSameIdExists(storeType: StoreType) {
+        val personList = createPersonsList(true)
+        val netStore = DataStore.collection(MULTI_INSERT_COLLECTION, Person::class.java, storeType, client)
+        clearBackend(netStore)
+        var saveCallback = createList(netStore, personList)
+        assertNull(saveCallback.error)
+        assertNotNull(saveCallback.result)
+        Assert.assertTrue(checkPersonIfSameObjects(personList, saveCallback.result?.entities, false))
+        saveCallback = createList(netStore, personList)
+        assertNull(saveCallback.error)
+        assertNotNull(saveCallback.result?.errors)
+        assertNotNull(saveCallback.result?.errors?.get(0)?.description)
+        assertNotNull(saveCallback.result?.errors?.get(0)?.debug)
```

Comment:
In general, I'd suggest to verify not just the field existence, but also the exact string, or at least some parts of it. For example, in this case the `description/debug` strings would contain something like "already exists" or "duplicate id', which we can compare against. 

I guess this is not done at the moment because theoreticaly the messages can be changed in the backend, but this is not something we usually do. And if we change something like this, it should be in a new API version. So the old one should continue to work as before, unless deprecated entirely.

Reply:
Fixed